File: services/pagesService.js
import { pagesRepository } from "@/repositories/pagesRepository";

export const pagesService = {

    async findBySearch(searchString, language) {
        try {
            const results = await pagesRepository.findBySearch(searchString, language);
            return Array.isArray(results) ? results : [];
        } catch(error) {
            console.error('Service layer search error:', error)
            return [];
        }
    }

}

File: services/pages_service.rb
class PagesService
  class << self
    def find_by_search(search_string, language)
      results = PagesRepository.find_by_search(search_string, language)
      results.is_a?(Array) ? results : []
    rescue StandardError => e
      Rails.logger.error("Service layer search error: #{e.message}")
      []
    end
  end
end
